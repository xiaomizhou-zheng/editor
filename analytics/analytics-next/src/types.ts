import { AnalyticsEventPayload } from './events/AnalyticsEvent';
import UIAnalyticsEvent from './events/UIAnalyticsEvent';

export type CreateUIAnalyticsEvent = (
  payload: AnalyticsEventPayload,
) => UIAnalyticsEvent;

export type AnalyticsEventCreator = (
  create: CreateUIAnalyticsEvent,
  props: Record<string, any>,
) => UIAnalyticsEvent | undefined;

export type CreateEventMapValue = AnalyticsEventPayload | AnalyticsEventCreator;

export type CreateEventMap = Record<string, CreateEventMapValue>;
