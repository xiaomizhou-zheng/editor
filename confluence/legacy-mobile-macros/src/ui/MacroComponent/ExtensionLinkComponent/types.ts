import { DocNode } from '@atlaskit/adf-schema';
import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface ExtensionLinkComponentProps {
  render: (document: DocNode) => JSX.Element;
  extension: ExtensionParams<any>;
}
