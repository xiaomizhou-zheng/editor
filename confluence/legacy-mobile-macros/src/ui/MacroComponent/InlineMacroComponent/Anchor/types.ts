import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface AnchorProps {
  extension: ExtensionParams<any>;
}
