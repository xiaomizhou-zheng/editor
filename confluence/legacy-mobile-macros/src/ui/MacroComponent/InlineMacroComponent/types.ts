import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface InlineMacroComponentProps {
  extension: ExtensionParams<any>;
}
