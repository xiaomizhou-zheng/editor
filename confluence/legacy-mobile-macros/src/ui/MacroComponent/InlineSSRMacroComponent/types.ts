import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export class CustomLegacyMacroError {
  constructor() {}
}

export interface CustomLegacyMacroResponse {
  html: string;
  resources: {
    css?: string;
    data?: string;
    js?: string;
  };
}

export interface InlineSSRMacroComponentProps {
  contentId: number;
  baseUrl: string;
  extension: ExtensionParams<any>;
  outputDeviceType: 'MOBILE' | 'DESKTOP';
  createPromise: Function;
  renderFallback: Function;
  onLinkClick: Function;
}
