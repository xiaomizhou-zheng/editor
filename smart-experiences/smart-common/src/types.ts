export enum EntityType {
  USER = 'USER',
  TEAM = 'TEAM',
  GROUP = 'GROUP',
}

export interface UserSearchBitbucketAttributes {
  /**
   * Identifies whether this is a public repository or not.
   */
  isPublicRepo?: boolean;
  /**
   * A list of bitbucket workspace Ids used within container result set and noted in analytics.
   */
  workspaceIds?: string[];
  /**
   * The users current email domain which may be used to boost the results for relevant users.
   */
  emailDomain?: string;
}

export interface UserSearchConfluenceAttributes {
  /**
   * Identifies whether this user is a confluence guest
   */
  isEntitledConfluenceExternalCollaborator?: boolean;
}

export type UserSearchProductAttributes =
  | UserSearchBitbucketAttributes
  | UserSearchConfluenceAttributes;

export interface UserSearchContext {
  childObjectId?: string;
  containerId?: string;
  contextType: string;
  objectId?: string;
  principalId?: string;
  productAttributes?: UserSearchProductAttributes;
  productKey: string;
  sessionId?: string;
  siteId: string;
}

export interface UserSearchQuery {
  cpusQueryHighlights?: {
    /**
     * The query used to highlight the results.
     */
    query?: string;
    /**
     * The fields to highlight.
     */
    field?: string;
  };
  /**
   * A lucene-style simple query formatted string for the term search expression. For example,
   * @example account_type:atlassian
   */
  customQuery?: string;
  /**
   * A customer directory ID to search for JSD customer accounts.
   */
  customerDirectoryId?: string;
  /**
   *  The lucene-style simple query format string for the filter (unscored) expression.
   *  Example:`account_status:"active" AND (NOT email_domain:"connect.atlassian.com")`
   *  will remove inactive users from the list of suggestions.
   */
  filter?: string;
  /**
   * Sets the minimum contextual access level that a user must have in order to be included in
   * the recommendations. Defaults to APPLICATION. At present, URS will not search for users with an
   * access level of NONE.
   */
  minimumAccessLevel?: string;
  /**
   * Restriction on the users and groups that should be applied to the search
   */
  restrictTo?: {
    userIds?: string[];
    groupIds?: string[];
  };
  /**
   * Whether the principal and the results require Jira application access or
   * just require site access without product access.
   */
  searchUserbase?: boolean;
}

export interface UserSearchRequest {
  baseUrl?: string;
  context: UserSearchContext;
  maxNumberOfResults?: number;
  performSearchQueryOnly?: boolean;
  query?: string;
  includeUsers?: boolean;
  includeGroups?: boolean;
  includeTeams?: boolean;
  searchQuery?: UserSearchQuery;
}

export interface UserSearchItem {
  id: string;
  entityType: EntityType;
  avatarUrl: string;
  description?: string;
  displayName?: string;
  name?: string;
  teamAri?: string;
}

export interface UserSearchResponse {
  errors?: any[];
  recommendedUsers: UserSearchItem[];
}
