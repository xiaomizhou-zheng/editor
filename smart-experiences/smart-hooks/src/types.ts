import type {
  UserSearchProductAttributes,
  UserSearchQuery,
} from '@atlaskit/smart-common';

export type UseUserRecommendationsProps = {
  /**
   * Base url override for downstream APIs.
   * Note: there is a trend for browsers blocking cookies for cross-site resources.
   * This prop is added for legacy reasons and will likely be removed in the future.
   * https://webkit.org/blog/10218/full-third-party-cookie-blocking-and-more/
   */
  baseUrl?: string;
  /**
   * Context information for analytics. Eg: if the hook was consumed inside a ticket's comment, the childObjectId would be
   *  the ID of the comment and the ticket would be the objectId. Optional, but please provide if available.
   */
  childObjectId?: string;
  /**
   * The container Id to identify context. e.g. Jira: projectId. Confluence: spaceId. Bitbucket: repositoryId.
   */
  containerId?: string;
  /**
   * Time to debounce the suggestions fetching (in milliseconds). Defaults to 150ms.
   */
  debounceTimeMs?: number;
  /**
   * Identifier for informing the server on what experience the hook is powering.
   * The server uses the fieldId to determine which model to utilize when
   * generating suggestions.
   * All fieldId's will be bucketed into a model that provides generic smart results,
   * except "assignee", "mentions" which are specifically trained for Jira Assignee and
   * @Mentions. For specifically trained models, please contact #help-smart-experiences.
   */
  fieldId: string;
  /**
   * Whether to include users in the result set  (@default: true).
   */
  includeUsers?: boolean;
  /**
   * Whether to include groups in the resultset (@default: false).
   */
  includeGroups?: boolean;
  /**
   * Whether to include teams in the resultset  (@default: false).
   */
  includeTeams?: boolean;
  /**
   *  (@default: 25)
   */
  maxNumberOfResults?: number;
  /**
   * An identifier of the closest context object, e.g. issueId, pageId, pullRequestId.
   * Used for analytics. Optional, but please include if available.
   */
  objectId?: string;
  /**
   * Preload the list of suggested users with an empty-query ('') search as soon as
   * the hook is used.
   * WARNING: please consider carefully before deciding to preload your suggestions
   * as this will increase the load on the recommendations services (has caused HOTs).
   * Please give #help-smart-experiences a ballpark on the expected request volume.
   * (@default: false)
   */
  preload?: boolean;
  /**
   * AAID of the user interacting with the component.
   * If not provided, server will extract principalId from the context header, assuming that the user is logged in
   * when making the request. (@default: “context”)
   */
  principalId?: string;
  /**
   * Product-specific Attributes - you should pass in the attribute type that matches your current SupportedProduct.
   * Currently we support additional attributes (BitbucketAttributes) for bitbucket and (ConfluenceAttributes) for Confluence.
   */
  productAttributes?: UserSearchProductAttributes;
  /**
   * Product identifier. If you are an NPF, please ensure your product has been onboarded with
   * Cross-product user-search @see https://developer.atlassian.com/cloud/cross-product-user-search/
   * If you are still waiting for CPUS, you can use the `people` productKey in the interim.
   */
  productKey: string;
  /**
   * Pass-through for CPUS. Refer to @see https://developer.atlassian.com/cloud/cross-product-user-search/rest/api-group-search/#api-group-search
   * for detailed docs on parameters.
   */
  cpusSearchQuery?: UserSearchQuery;
  /**
   * Identifier for the product's tenant, also known as siteId or cloudId.
   * If product is not tenanted, the convention is to provide the productKey
   * as the tenantId (e.g. "bitbucket") but this may vary across products.
   * Check with #help-smart-experiences.
   */
  tenantId: string;
};
