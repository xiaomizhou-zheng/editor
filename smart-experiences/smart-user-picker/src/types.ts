import { WithAnalyticsEventsProps } from '@atlaskit/analytics-next';
import {
  DefaultValue,
  OptionData,
  UserPickerProps,
} from '@atlaskit/user-picker';

export interface Context {
  containerId?: string;
  contextType: string;
  objectId?: string;
  sessionId?: string;
  principalId?: string;
  childObjectId?: string;
  productKey: string;
  siteId: string;
  organizationId?: string;
  productAttributes?: ProductAttributes;
}

export interface RecommendationRequest {
  baseUrl?: string;
  context: Context;
  maxNumberOfResults: number;
  query?: string;
  searchQueryFilter?: string;
  includeUsers?: boolean;
  includeGroups?: boolean;
  includeTeams?: boolean;
}

type OnError = (
  error: any,
  request: RecommendationRequest,
) => Promise<OptionData[]> | void;
type OnValueError = (
  error: any,
  defaultValue: DefaultValue,
) => Promise<OptionData[]> | void;
type OnEmpty = (query: string) => Promise<OptionData[]>;

export interface State {
  users: OptionData[];
  loading: boolean;
  closed: boolean;
  query: string;
  sessionId?: string;
  defaultValue?: DefaultValue;
  bootstrapOptions: OptionData[];
}

export type ProductAttributes = BitbucketAttributes | ConfluenceAttributes;

export type FilterOptions = (
  options: OptionData[],
  query: string,
) => OptionData[];

export interface BitbucketAttributes {
  /**
   * Identifies whether this is a public repository or not.
   */
  isPublicRepo?: boolean;
  /**
   * A list of bitbucket workspace Ids used within container result set and noted in analytics.
   */
  workspaceIds?: string[];
  /**
   * The users current email domain which may be used to boost the results for relevant users.
   */
  emailDomain?: string;
}

export interface ConfluenceAttributes {
  /**
   * Identifies whether this user is a confluence guest
   */
  isEntitledConfluenceExternalCollaborator?: boolean;
}

export enum EntityType {
  USER = 'USER',
  TEAM = 'TEAM',
  GROUP = 'GROUP',
}

export interface RecommendationItem {
  id: string;
  name?: string;
  entityType: EntityType;
  avatarUrl: string;
  description?: string;
  teamAri?: string;
  displayName?: string;
}

export interface RecommendationResponse {
  errors?: any[];
  recommendedUsers: RecommendationItem[];
}

export interface SmartProps {
  /**
   * The base URL of the site eg: hello.atlassian.net
   */
  baseUrl?: string;
  /**
   * Hydrated user suggestions to show when the query is blank. If not provided, smart user picker
   *  will still provide a smart-ranked list of suggestions for blank queries. Please refer to @atlaskit/user-picker
   *  for OptionData type.
   */
  bootstrapOptions?: OptionData[];
  /**
   * Context information for analytics. Eg: if a user picker was put inside a comment, the childObjectId would be
   *  the ID of the comment. Optional, but please provide if available.
   */
  childObjectId?: string;
  /**
   * The container Id to identify context.
   *
   * e.g. Jira: projectId. Confluence: spaceId. Bitbucket: repositoryId.
   */
  containerId?: string;
  /**
   * Time to debounce the suggestions fetching (in milliseconds). Defaults to 150ms.
   */
  debounceTime?: number;
  /**
   * Function to transform options suggested by the server before showing to the user. Can be used to filter out suggestions.
   * The results of filterOptions are the results displayed in the suggestions UI.
   */
  filterOptions?: FilterOptions;
  /**
   * Whether to include groups in the resultset. Only supported for Confluence. @default false
   */
  includeGroups?: boolean;
  /**
   * Whether to include teams in the resultset. @default false
   */
  includeTeams?: boolean;
  /**
   * Whether to include users in the resultset. @default true
   */
  includeUsers?: boolean;
  /**
   * An identifier of the closest context object, e.g. issueId, pageId, pullRequestId.
   * Used for analytics. Optional, but please include if available.
   */
  objectId?: string;
  /**
   * Custom handler to give opportunity for caller to return list of options when server returns empty list.
   * this is called if server returns empty list. This will NOT be called if props.filterOptions returns empty list.
   */
  onEmpty?: OnEmpty;
  /**
   * Error handler for when the server fails to suggest users and returns with an error response.
   * `error`: the error.
   * `RecommendationRequest`: the original recommendationRequest containing the query and other search parameters.
   * This may be used to provide a fail over search direct to the product backend.
   * Helper fail over clients exist under /helpers.
   * Note that OnError results are filtered.
   */
  onError?: OnError;
  /**
   * Error handler used to provide OptionData[] values when the server fails to hydrate the `defaultValue` prop's values.
   */
  onValueError?: OnValueError;
  /**
   * Prefetch the list of suggested assignees before the user picker is focused.
   * WARNING: please consider carefully before deciding to prefetch your suggestions
   * as this will increase the load on the recommendations services (has caused HOTs).
   * Please give #help-smart-experiences a ballpark on the expected request volume.
   */
  prefetch?: boolean;
  /**
   * Id of the user interacting with the component.
   * If principalId is not provided, server will extract principalId from the context header, assuming that the user is logged in
   *  when making the request. @default “context”
   */
  principalId?: string;
  /**
   * Product-specific Attributes - you should pass in the attribute type that matches your current SupportedProduct.
   * Currently we support additional attributes (BitbucketAttributes) for bitbucket and (ConfluenceAttributes) for Confluence.
   */
  productAttributes?: ProductAttributes;
  /**
   * Product identifier. If you are an NPF, please ensure your product has been onboarded with
   *  Cross-product user-search @see https://developer.atlassian.com/cloud/cross-product-user-search/
   * If you are still waiting for CPUS, you can use the `people` productKey in the interim.
   */
  productKey: string;
  /**
   * Filter to be applied to the eventual query to CPUS for user suggestions.
   * Example:`account_status:"active" AND (NOT email_domain:"connect.atlassian.com")`
   *  will remove inactive users from the list of suggestions.
   */
  searchQueryFilter?: string;
  /**
   * Identifier for the product's tenant, also known as tenantId or cloudId
   */
  siteId: string;
  /**
   * Identifier for the organization in which to search for teams.
   */
  orgId?: string;
}

// Override UserPickerProps below with replacement documentation
export interface Props
  extends SmartProps,
    UserPickerProps,
    WithAnalyticsEventsProps {
  /**
   * The pre-selected values for the smart user picker. Supports only Users and Teams default value hydration.
   * If the `DefaultValue` contains only an `id` and `type` (it conforms to an `OptionIdentifier`)
   * then the values will be automatically hydrated.
   * If the value has a `name` then it is considered hydrated and will be ignored.
   * Uses Confluence and Jira if called from there, else uses Identity or Legion for teams. If a value could not be found, or there was
   * a network failure during the hydration, the value will be rendered with the label 'Unknown'. Else, if there were any other error
   * during default value hydration, no default values will be rendered, use `onValueError` to handle this.
   * `defaultValue` differs from `value` in that it sets the initial value then leaves the component 'uncontrolled'
   * whereas setting the `value` prop delegates responsibility for maintaining the value to the caller
   * (i.e. listen to `onChange`)
   */
  defaultValue?: DefaultValue;
  /**
   * Identifier for informing the server on where the user picker has been mounted.
   * Unlike User Picker, the fieldId in Smart User Picker is mandatory.
   * The server uses the fieldId to determine which model to utilize when
   * generating suggestions.
   * All fieldId's will be bucketed into a model that provides generic smart results,
   * except "assignee", "mentions" which are specifically trained for Jira Assignee and
   * @Mentions. For specifically trained models, please contact #help-smart-experiences.
   */
  fieldId: string;
}
