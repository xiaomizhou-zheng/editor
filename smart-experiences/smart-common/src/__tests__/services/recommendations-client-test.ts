import fetchMock from 'fetch-mock/cjs/client';

import fetchUserRecommendations from '../../services/recommendations-client';
import { UserSearchQuery, UserSearchRequest } from '../../types';

const URS_URL = '/gateway/api/v1/recommendations';

const exampleContext = {
  childObjectId: 'childObjectId',
  containerId: 'containerId',
  contextType: 'fieldId',
  objectId: 'objectId',
  principalId: 'principalId',
  productAttributes: {
    emailDomain: 'Context',
    isPublicRepo: false,
    workspaceIds: ['02b941e3-cfaa-40f9-9a58-cec53e20bdc3'],
    pullRequestId: 18056,
  },
  productKey: 'bitbucket',
  sessionId: 'session-id',
  siteId: 'site-id',
};

const exampleCpusSearchQuery: UserSearchQuery = {
  cpusQueryHighlights: {
    query: 'query',
    field: 'field',
  },
  customQuery:
    'name.mention:test_user email.mention:test_user nickname.mention:test_user',
  customerDirectoryId: '37e1dc69-9db4-4d1e-8083-a663329ba939',
  filter: '(NOT not_mentionable:true) AND (account_status:active)',
  minimumAccessLevel: 'APPLICATION',
  restrictTo: {
    userIds: ['user1', 'user2'],
    groupIds: ['group1', 'group2'],
  },
  searchUserbase: true,
};

const exampleRequest: UserSearchRequest = {
  context: exampleContext,
  maxNumberOfResults: 50,
  query: 'query',
  searchQuery: exampleCpusSearchQuery,
  includeUsers: true,
  includeGroups: true,
  includeTeams: true,
};

const exampleResponse = {
  status: 200,
  body: JSON.stringify({
    recommendedUsers: [
      {
        entityType: 'USER',
        id: '5ee0adc79583380ab0afec31',
        name: 'John Smith',
        email: 'jsmith@atlassian.com',
        avatarUrl:
          'https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net',
        nickname: 'John Smith',
        matchPositions: {},
        accessLevel: 'CONTAINER',
        accountStatus: 'ACTIVE',
        notMentionable: false,
        locale: 'en-GB',
        title: 'Development Manager',
        userType: 'DEFAULT',
      },
      {
        entityType: 'USER',
        id: '5ee0adc79583380ab0afec32',
        name: 'John Doe',
        email: 'jsmith@atlassian.com',
        avatarUrl:
          'https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net',
        nickname: 'John Doe',
        matchPositions: {},
        accessLevel: 'CONTAINER',
        accountStatus: 'ACTIVE',
        notMentionable: false,
        locale: 'en-GB',
        title: 'Development Manager',
        userType: 'DEFAULT',
      },
    ],
  }),
};

describe('default-value-hydration-client', () => {
  afterEach(() => {
    jest.clearAllMocks();
    fetchMock.restore();
  });

  it('should reject promise on network error', async () => {
    fetchMock.post(URS_URL, 504);

    try {
      await fetchUserRecommendations(exampleRequest);
    } catch (error) {
      expect((error as any).message).toMatchSnapshot('URS error');
    }
  });

  it('should translate given context to correct fetch request', async () => {
    let requestBody;

    fetchMock.post(
      {
        functionMatcher: (url: string, options: any) => {
          requestBody = JSON.parse(options.body);

          return url === '/gateway/api/v1/recommendations';
        },
      },
      exampleResponse,
      {
        repeat: 1,
        overwriteRoutes: false,
      },
    );

    const users = await fetchUserRecommendations(exampleRequest);

    expect(fetchMock.called()).toBeTruthy();
    expect(requestBody).toMatchSnapshot('URS query');
    expect(users).toMatchSnapshot('URS users');
  });
});
