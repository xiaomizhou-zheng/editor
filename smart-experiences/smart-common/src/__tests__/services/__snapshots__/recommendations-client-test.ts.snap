// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-value-hydration-client should reject promise on network error: URS error 1`] = `"error calling smart service, statusCode=504, statusText=Gateway Timeout"`;

exports[`default-value-hydration-client should translate given context to correct fetch request: URS query 1`] = `
Object {
  "context": Object {
    "childObjectId": "childObjectId",
    "containerId": "containerId",
    "contextType": "fieldId",
    "objectId": "objectId",
    "principalId": "principalId",
    "productAttributes": Object {
      "emailDomain": "Context",
      "isPublicRepo": false,
      "pullRequestId": 18056,
      "workspaceIds": Array [
        "02b941e3-cfaa-40f9-9a58-cec53e20bdc3",
      ],
    },
    "productKey": "bitbucket",
    "sessionId": "session-id",
    "siteId": "site-id",
  },
  "includeGroups": true,
  "includeTeams": true,
  "includeUsers": true,
  "maxNumberOfResults": 50,
  "searchQuery": Object {
    "cpusQueryHighlights": Object {
      "field": "field",
      "query": "query",
    },
    "customQuery": "name.mention:test_user email.mention:test_user nickname.mention:test_user",
    "customerDirectoryId": "37e1dc69-9db4-4d1e-8083-a663329ba939",
    "filter": "(NOT not_mentionable:true) AND (account_status:active)",
    "minimumAccessLevel": "APPLICATION",
    "queryString": "query",
    "restrictTo": Object {
      "groupIds": Array [
        "user1",
        "user2",
      ],
      "userIds": Array [
        "user1",
        "user2",
      ],
    },
    "searchUserbase": true,
  },
}
`;

exports[`default-value-hydration-client should translate given context to correct fetch request: URS users 1`] = `
Array [
  Object {
    "accessLevel": "CONTAINER",
    "accountStatus": "ACTIVE",
    "avatarUrl": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net",
    "email": "jsmith@atlassian.com",
    "entityType": "USER",
    "id": "5ee0adc79583380ab0afec31",
    "locale": "en-GB",
    "matchPositions": Object {},
    "name": "John Smith",
    "nickname": "John Smith",
    "notMentionable": false,
    "title": "Development Manager",
    "userType": "DEFAULT",
  },
  Object {
    "accessLevel": "CONTAINER",
    "accountStatus": "ACTIVE",
    "avatarUrl": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net",
    "email": "jsmith@atlassian.com",
    "entityType": "USER",
    "id": "5ee0adc79583380ab0afec32",
    "locale": "en-GB",
    "matchPositions": Object {},
    "name": "John Doe",
    "nickname": "John Doe",
    "notMentionable": false,
    "title": "Development Manager",
    "userType": "DEFAULT",
  },
]
`;
